snapshot/testsuite/testsuite.go

```diff
@@ -182,13 +182,13 @@ func checkSnapshotterBasic(ctx context.Context, t *testing.T, snapshotter snapsh
 	if err := mount.MountAll(mounts, nextnext); err != nil {
 		t.Fatalf("failure reason: %+v", err)
 	}
-	defer testutil.Unmount(t, nextnext)
 
 	if err := fstest.CheckDirectoryEqualWithApplier(nextnext,
 		fstest.Apply(initialApplier, diffApplier)); err != nil {
 		t.Fatalf("failure reason: %+v", err)
 	}
 
+	testutil.Unmount(t, nextnext)
```

Comment:
Don't you want to call `unmount` if `fstest.CheckDirectoryEqualWithApplier` fails too?

What about making the `Remove` in the assert deferred called instead (i.e after the View/Commit calls they're related to)?